darwin/dataset/remote_dataset.py

```diff
@@ -106,60 +117,39 @@ def push(
-
-        # This is where the responses from the upload function will be saved/load for resume
-        self.local_path.parent.mkdir(exist_ok=True)
-        responses_path = self.local_path.parent / ".upload_responses.json"
         # Init optional parameters
         if files_to_exclude is None:
             files_to_exclude = []
         if files_to_upload is None:
-            raise NotFound("Dataset location not found. Check your path.")
-
-        if resume:
-            if not responses_path.exists():
-                raise NotFound("Dataset location not found. Check your path.")
-            with responses_path.open() as f:
-                logged_responses = json.load(f)
-            files_to_exclude.extend(
-                [
-                    response["file_path"]
-                    for response in logged_responses
-                    if response["s3_response_status_code"].startswith("2")
-                ]
-            )
+            raise ValueError("No files or directory specified.")
 
         files_to_upload = find_files(files=files_to_upload, recursive=True, files_to_exclude=files_to_exclude)
 
         if not files_to_upload:
             raise ValueError("No files to upload, check your path, exclusion filters and resume flag")
 
-        progress, count = add_files_to_dataset(
-            client=self.client,
-            dataset_id=str(self.dataset_id),
-            filenames=files_to_upload,
-            fps=fps,
-            as_frames=as_frames,
-            team=self.team,
-            path=path,
-        )
+        local_files = []
+        for file in files_to_upload:
+            local_files.append(LocalFile(file, fps=fps, as_frames=as_frames, path=path))
+
+        handler = UploadHandler(self.client, local_files, DatasetIdentifier(self.slug, self.team))
+        handler.upload()
 
         # If blocking is selected, upload the dataset remotely
-        if blocking:
-            responses = exhaust_generator(progress=progress, count=count, multi_threaded=multi_threaded)
-            # Log responses to file
-            if responses:
-                responses = [{k: str(v) for k, v in response.items()} for response in responses]
-                if resume:
-                    responses.extend(logged_responses)
-                with responses_path.open("w") as f:
-                    json.dump(responses, f)
-            return None, count
-        else:
-            return progress, count
+        if blocking and handler.pending_count:
+            import concurrent.futures
+            with concurrent.futures.ThreadPoolExecutor(max_workers=5) as executor:
+                future_to_progress = {executor.submit(f): f for f in handler.progress}
+                for future in concurrent.futures.as_completed(future_to_progress):
+                    try:
+                        future.result()
+                    except Exception as exc:
+                        print('%r generated an exception: %s' % (url, exc))
```

Comment:
There are a lot of `print` in the code, what if we use the [logging module](https://docs.python.org/3/howto/logging.html), so we can also persist `cli` logs for better debugging/support.